code/game/machinery/status_display.dm

```diff
@@ -248,37 +248,7 @@
 		return
 
 	if(mode==1)	// AI emoticon
-		switch(emotion)
-			if("Very Happy")
-				set_picture("ai_veryhappy")
-			if("Happy")
-				set_picture("ai_happy")
-			if("Neutral")
-				set_picture("ai_neutral")
-			if("Unsure")
-				set_picture("ai_unsure")
-			if("Confused")
-				set_picture("ai_confused")
-			if("Sad")
-				set_picture("ai_sad")
-			if("BSOD")
-				set_picture("ai_bsod")
-			if("Blank")
-				set_picture("ai_off")
-			if("Problems?")
-				set_picture("ai_trollface")
-			if("Awesome")
-				set_picture("ai_awesome")
-			if("Dorfy")
-				set_picture("ai_urist")
-			if("Facepalm")
-				set_picture("ai_facepalm")
-			if("Friend Computer")
-				set_picture("ai_friend")
-			if("Blue Glow")
-				set_picture("ai_sal")
-			if("Red Glow")
-				set_picture("ai_hal")
+		set_picture("ai_[emotion]")
```

Comment:
I'm not sure if the string is case sensitive, but if it is you might want to lower all the text of emotion.

Also "Problems?" will give you "ai_problems?" which I don't think is a valid picture? I think maybe change "Problems?" prompt to be "trollface" or the other way around.

And I can see some other unique options like "Very Happy" needs to become "veryhappy". Replace all spaces with blank string? Then there's "friend computer", "Dorfy", "Blue glow", and "red glow", none of these are valid unless I'm mistaken